menu.sh

```diff
@@ -371,7 +397,8 @@ case $mainmenu_selection in
 	else
 		echo "Install Docker"
 		curl -fsSL https://get.docker.com | sh
-		sudo usermod -aG docker $USER
+		sudo usermod -G docker -a $USER
+		sudo usermod -G bluetooth -a $USER
```

Comment:
Does the bluetooth group exist on all systems?

Reply:
Good question. I only have RPi3B+ and RPi4 and it exists there. It didn't occur to me to wonder about anything else. Perhaps I can invert the question and ask if you know of IOTstack targets that don't have bluetooth?

Reply:
Having tried this with:

```
sudo usermod -G doesnotexist -a $USER
```

all you get is a warning so it would not matter for that part of the PR.

But it **would** probably cause a problem where user\_in\_group() is used because, if "bluetooth" did not exist, the user would get a reboot prompt on every menu run. Not harmful but annoying.

After some experimentation, here's a test script that carries a revised user\_in\_group():

```
#!/bin/bash

function user_in_group()
{

   # see if the group exists
   grep -q "^$1:" /etc/group;

   # sense that the group does not exist
   if [ $? -ne 0 ]; then return 0; fi

   # group exists - now check that the user is a member
   groups | grep -q "\b$1\b"

}

if user_in_group $1 ; then
   echo "checked membership of $1 group - the script would not need to repair anything"
else
   echo "the group $1 exists but the user is NOT a member - the script would need to repair this"
fi
```

Tests:

* RPi4: group=docker exists, user already a member:

	```
	$ ./user_in_group_test docker
	checked membership of docker group - the script would not need to repair anything
	```

* RPi4: group=bluetooth exists, user already a member:

	```
	$ ./user_in_group_test bluetooth
	checked membership of bluetooth group - the script would not need to repair anything
	```
* RPi4: group=rdma exists, user **not** a member:

	```
	$ ./user_in_group_test rdma
	the group rdma exists but the user is NOT a member - the script would need to repair this
	```

* RPi4: group=nonexistent does not exist, membership not possible:

	```
	$ ./user_in_group_test nonexistent
	checked membership of nonexistent group - the script would not need to repair anything
	```

* RPi3 (a **different** machine): group=bluetooth exists, user **not** a member

	```
	$ ./user_in_group_test bluetooth
	the group bluetooth exists but the user is NOT a member - the script would need to repair this
	```

If you agree that the revised user\_in\_group() would be preferable, please go ahead and implement it like that. This is another way of me saying that if you flick the problem back to me to fix the PR, that's actually beyond my git skills so I would need guidance.

Reply:
Yeah, I would need access to your repo in order to push changes to this PR (I think you did grant them in the past though).

I'm not sure how you're committing, but basically, if you are currently on this branch, make your changes to the code and then just run:
```
git add .
git commit
git push
```
If git push is complaining, it probably wants your repos name, and the branch name. I'm not sure what you called it on your local system, but I usuaully name it `fork` or something, so:
```
git push fork 20200615-nodered-bluetooth
```

I would suggest learning how git works on the CLI, but if you can't be bothered, or just don't have the time, checkout [SourceTree](https://www.sourcetreeapp.com/), a true blue Aussie invention my Atlassian. It has a great UI for a lot of the common features you'd use on the command line.

I gotta sleep and then go to work, will test out the rest of this PR shortly!

Reply:
I did it this way:

```
# prepare as for making PR (checkout the fork, add the upstream)
git checkout 20200615-nodered-bluetooth
vi menu.sh 
git add menu.sh 
git commit -a -S
git push --signed=if-asked origin 20200615-nodered-bluetooth
```

The forked repo on the web didn't tell me to do anything else (like it does when I first push a new branch to become a PR) and the revisions are there so my guess is it's happy.

Learn something new every day.

Oh, I'm a die-hard CLI person. Never been a big fan of GUIs where there's a CLI. I just live in fear of making a major cock-up that creates work for other people.